internal/statemachine/helper.go

```diff
@@ -1012,42 +1022,121 @@ func (stateMachine *StateMachine) updateGrub(rootfsVolName string, rootfsPartNum
 			relpath:  "/sys",
 			typ:      "sysfs",
 		},
+		{
+			basePath: mountDir,
+			relpath:  "/run",
+			bind:     true,
+		},
 	}
 
-	for _, mp := range mountPoints {
-		mountCmds, umountCmds, err := mp.getMountCmd()
-		if err != nil {
-			return fmt.Errorf("Error preparing mountpoint \"%s\": \"%s\"",
-				mp.relpath,
-				err.Error(),
-			)
-		}
-		updateGrubCmds = append(updateGrubCmds, mountCmds...)
-		teardownCmds = append(umountCmds, teardownCmds...)
+	mountCmds, umountCmds, err := generateMountPointCmds(mountPoints, stateMachine.tempDirs.scratch)
+	if err != nil {
+		return err
 	}
 
+	setupGrubCmds = append(setupGrubCmds, mountCmds...)
+	teardownCmds = append(umountCmds, teardownCmds...)
+
 	teardownCmds = append([]*exec.Cmd{
 		execCommand("udevadm", "settle"),
 	}, teardownCmds...)
 
+	// udev needed to have grub-install properly work
```

Comment:
Without it it failed to run in a `minbase` chroot. See the `TestPackStateMachine_SuccessfulRun` test and previous comment by @kukrimate who found the fix.
But if you we could avoid installing it I am interested :).

Reply:
That test debootstraps jammy so that tracks. Given that I would expect the majority of ubuntu-image's usage to be for releases newer than jammy (to put it mildly) I would prefer not to have this cruft here...

Reply:
AFAIK some of our users in other teams still build jammy-based images. I would like to discuss our policy regarding support of building old images.

Reply:
Well OK can we say "if release <= jammy" somewhere?

I got distracted halfway through responding to your comments, sorry about that. But I guess we're talking about all this stuff in a few hours anyway.

Reply:
So far I have been doing everything I could to avoid hardcoding series-specific stuff in the tool. I would rather understand the root cause and find an heuristic to determine if installing udev is needed or not. Do you know why it was needed for jammy and older (and why it is not anymore?).